core/src/main/java/com/rudderstack/android/sdk/core/FlushUtils.java

```diff
@@ -0,0 +1,244 @@
+package com.rudderstack.android.sdk.core;
+
+import android.text.TextUtils;
+
+import androidx.annotation.Nullable;
+
+import com.rudderstack.android.sdk.core.util.Utils;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Standard flush related calls
+ */
+class FlushUtils {
+    private FlushUtils() {
+
+    }
+
+    private static final Object FLUSH_LOCK = new Object();
+    private static final Object DB_LOCK = new Object();
+
+    /**
+     * Should not be called from main thread.
+     *
+     * @param areFactoriesInitialized
+     * @param integrationOperationsMap
+     * @param messageIds
+     * @param messages
+     * @param flushQueSize
+     * @param dataPlaneUrl
+     * @param dbManager
+     * @param authHeaderString
+     * @param anonymousIdHeaderString
+     * @return
+     */
+    static boolean flush(boolean areFactoriesInitialized, @Nullable Map<String, RudderIntegration<?>> integrationOperationsMap,
+                         List<Integer> messageIds, List<String> messages, int flushQueSize, String dataPlaneUrl,
```

Comment:
Nitpick - Feel free to ignore. 

can we name this flushQueueSize instead? Or is it a standard practice to name it that way?